### ISSUE TITLE
No detections on custom data training

### ISSUE THREAD

**Issue by Just1813** [ultralytics/yolov5#12990]
### Search before asking

- [ ] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I'm trying to train yolov5 with custom data. I'm using very few images just to test everything. I have 2 classes and 6 images (3 images for each class), which I know is WAY too little, but as I said, it's only for testing purposes. So I trained the model with the data which worked fine, but when I try to detect one of the images that was used for training, it doesn't work. Shouldn't it be able to detect the images I trained with, even if there are so few?

Command for training:
python train.py --img 2048 --batch 16 --epochs 5 --data test.yaml --weights yolov5s.pt --nosave --cache

Command for testing:
python detect.py --weights runs/train/exp11/weights/last.pt --source data/cartes_mini/images/2-C_jpg.rf.3ad5f752441ac8389c42afec2b5ecc10.jpg

I've also tried with another dataset that contained 1 class and around 20 training images, but got the same result.

Thank you!

**Comment by glenn-jocher** [comment]
Hello! Thanks for reaching out. 👋

Absolutely, even with a small dataset, you'd typically expect some detections from the training images. A few potential checkpoints:

1. **Annotations Check**: Ensure the annotations are correctly formatted and correspond accurately to your class labels in the images. Incorrect or misaligned annotations could result in no detections.

2. **Image Path**: Verify that the image paths specified in your data YAML and testing command are correct and accessible.

3. **Experimentation**: Try to lower the confidence threshold when running the detection command to see if the model is actually detecting anything but with low confidence:

```python
python detect.py --weights runs/train/exp11/weights/last.pt --source data/cartes_mini/images/2-C_jpg.rf.3ad5f752441ac8389c42afec2b5ecc10.jpg --conf 0.25
```

4. **Model Overfitting**: With such a small dataset, the model might be underfitting. Consider using pre-trained weights and fine-tuning on your data for a better generalization over few epochs.

If you're persistently encountering issues, consider adjusting training parameters or increasing dataset size for effective training results. Happy coding! 😊
